Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -622,9 +613,12 @@ class Navigator {
                                   << surface->geometryId());
       // Estimate the surface status
       bool boundaryCheck = true;
-      if (std::find(externalSurfaces.begin(), externalSurfaces.end(),
-                    surface->geometryId()) != externalSurfaces.end()) {
-        boundaryCheck = false;
```

Comment:
I do not understand why the search did not work but the loop over the external surfaces does ... can you explain, please?

Reply:
If I remember correctly, for this change both worked. This change is just to make it faster, I think the reasoning I had was that  since `state.navigation.externalSurfaces` is a multimap I couldn't use find with it directly so I had to fill a vector then used `std::find` with the vector. In the new version I just do a loop over the range so I don't have to do the copy.

Reply:
ok, fine with me then.

Reply:
Sorry I missed your question. Yes that is the idea, to be honest the `externalSurfaces` were already there as a multimap just not implemented but I found it convenient that way so I kept them as is.